### PR TITLE
[EventGrid] Migrate track2 SDK

### PR DESCRIPTION
**Pull request by 00Kai0**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/eventgrid/_params.py [hunk 1]
```diff
@@ -233,6 +233,7 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
 
     with self.argument_context('eventgrid topic key') as c:
         c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('key_name', help='Key name to regenerate key1 or key2')
```
Comment (ahamad-MS) [inline src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:236]: Why is this needed/added? Same for other key arguments for domains and partner namespaces

Comment (ahamad-MS) [reply src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:236]: never mind.. I got it.

**Review by ahamad-MS — APPROVED** [review]
LGTM.

**Review by Juliehzl — APPROVED** [review]
